.github/workflows/build.yml

```diff
@@ -0,0 +1,73 @@
+name: "Build"
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+    branches:
+      - develop
+  push:
+    branches:
+      - develop
+
+jobs:
+  tests:
+    name: Tests
+    uses: ./.github/workflows/test.yml
+
+  generate-build-id:
+    name: "Generate Build Id"
+    needs: [ tests ]
+    runs-on: ubuntu-latest
+    outputs:
+      build-id: ${{ steps.generate.outputs.buildId }}
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - id: generate
+        working-directory: ./scripts
+        shell: bash
+        env:
+          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          chmod +x ./create_build_id.sh
+
+          if [[ "${{ github.event_name }}" == "pull_request" ]]; then
+          GIT_BRANCH=PR
+          elif [[ "${{ github.event_name }}" == "push" && "${{ github.ref }}" == "refs/heads/main" ]]; then
+          GIT_BRANCH=main
+          fi
+
+          BUILD_ID=$(./create_build_id.sh $GIT_BRANCH ${{ github.run_number }} ${{ github.sha }})
+          echo "Generated the build tag: $BUILD_ID"
+          echo "buildId=$BUILD_ID" >> "$GITHUB_OUTPUT"
+
+  publish-docker-image:
+    name: "Publish docker image to ECR"
+    needs: [ generate-build-id ]
+
+    uses: ./.github/workflows/publish.yml
+    with:
+      directory: .
+      repository: nhais
+      build-context: .
+      build-id: ${{ needs.generate-build-id.outputs.build-id }}
+    secrets: inherit
+
+  comment:
+    if: github.event_name == 'pull_request'
+    name: "Create Build ID Comment"
+    needs: [ generate-build-id, publish-docker-image]
+    continue-on-error: true
+    permissions: write-all
+    runs-on: [ ubuntu-latest ]
```

Comment:
I don't think we need the squire brackets here especially that before we didn't use them on line 19

Reply:
yeah we shouldn't need them, it makes no odds, but I'll remove them for clarity